lib/em-hiredis/pubsub_client.rb

```diff
@@ -248,10 +248,11 @@ def unsubscribe_impl(type, subscriptions, channel)
     end
 
     def unsubscribe_proc_impl(type, subscriptions, channel, proc)
-      removed = subscriptions[channel].delete(proc)
+      return unless subscriptions.include?(channel)
```

Comment:
I agree that `#unsubscribe_impl` could also benefit from an early-return. The others either have multiple meaty branches or a single-line condition.